sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -37,15 +37,23 @@ trait LimitExec extends UnaryExecNode {
 }
 
 /**
- * Take the first `limit` elements and collect them to a single partition.
+ * Take the first `limit` + `offset` elements and collect them to a single partition and then to
+ * drop the first `offset` elements.
  *
  * This operator will be used when a logical `Limit` operation is the final operator in an
  * logical plan, which happens when the user is collecting results back to the driver.
  */
-case class CollectLimitExec(limit: Int, child: SparkPlan) extends LimitExec {
+case class CollectLimitExec(limit: Int, offset: Int, child: SparkPlan) extends LimitExec {
   override def output: Seq[Attribute] = child.output
   override def outputPartitioning: Partitioning = SinglePartition
-  override def executeCollect(): Array[InternalRow] = child.executeTake(limit)
+  override def executeCollect(): Array[InternalRow] = {
+    // Because CollectLimitExec collect all the output of child to a single partition, so we need
+    // collect the first `limit` + `offset` elements and then to drop the first `offset` elements.
+    // For example: limit is 1 and offset is 2 and the child output two partition.
+    // The first partition output [1, 2, 3] and the Second partition output [4, 5].
+    // Then [1, 2, 3] or [4, 5, 1] will be taken and output [3] or [1].
```

Comment:
`[1, 2, 3] or [4, 5, 1]` are you sure? AFAIK `RDD.take` will collect the results w.r.t. the partition order, so it should always be `[1, 2, 3]`